CLN: parametrize test_nat_comparisons

ivanovmg opened the pull request:
Parametrize ``TestDatetime64SeriesComparison.test_nat_comparisons``
using ``operator`` module.

If this kind of cleanup is considered reasonable,
then I will go through the remaining tests in this test module.

- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

ivanovmg committed:
CLN: parametrize test_nat_comparisons

Parametrize TestDatetime64SeriesComparison.test_nat_comparisons
using ``operator`` module.

jreback commented:
yes this type of cleanup is good. targeted PRs are the best.

cc @jbrockmendel if comments.

jbrockmendel commented:
i do not find this clearer than the status quo.

the lowest-hanging fruit in the tests/arithmetic directory are ones that do not use the box_with_array fixture

there are also very likely redundant tests that can be de-duplicated

jreback commented:
@jbrockmendel ok so you are -1 on this?

jbrockmendel commented:
-0.5, i.e. i dont think this is an improvement, but not a hill to die on if others have strong opinions

jreback commented:
ok closing this

simonjayhawkins commented:
> -0.5, i.e. i dont think this is an improvement, but not a hill to die on if others have strong opinions

Parameterisation is beneficial when there a multiple test failures. It is sometimes easier to debug when all the passing and failing test cases are visible together.

This PR looked good to me.

jreback commented:
ok @ivanovmg can u merge master

ivanovmg committed:
Merge branch 'master' into cleanup/arithmetic

ivanovmg committed:
REF: rename operator variable to op

Rename local variable operator -> op.
This is done for consistency with other tests in the module.
And also not to confuse with ``operator`` module.

ivanovmg commented:
Merged master.
Also I renamed local variable ``operator`` into ``op`` for consistency with other test and not to confuse with module ``operator``.
However, probably more readable option would be ``operator_`` with a trailing underscore, so it is more clear that we are talking about the operator.
Alternatively, ``binary_operator`` - longer name, but more clear IMHO.

jbrockmendel commented:
> Parameterisation is beneficial when there a multiple test failures. It is sometimes easier to debug when all the passing and failing test cases are visible together.

This is true in a general sense, but I'd point out:

1) we have many tests where we don't parametrize these particular 6 methods

2) The 6 Series objects get created at test collection time, which increases the memory footprint of the test suite.  In isolation these 6 small Series is not a big deal, but we are already at a point where memory is causing the py38 windows build to fail regularly.

Again, not a hill worth dying on, just saying.

jbrockmendel commented on pandas/tests/arithmetic/test_datetime64.py:
```diff
@@ -152,8 +152,6 @@ def test_dt64arr_nat_comparison(self, tz_naive_fixture, box_with_array):
 
 
 class TestDatetime64SeriesComparison:
-    # TODO: moved from tests.series.test_operators; needs cleanup
```
this comment pertains to the entire class, not just this test

ivanovmg replied:
Got it. Moved it back.

ivanovmg committed:
FIX: Add removed comment

jreback commented:
thanks @ivanovmg
